src/R/Components/Impl/Plots/Implementation/View/RPlotDeviceControl.xaml.cs

```diff
@@ -0,0 +1,113 @@
+
+            // If the window gets below a certain minimum size, plot to the minimum size
+            int pixelWidth = Math.Max((int)unadjustedPixelSize.Width, MinPixelWidth);
+            int pixelHeight = Math.Max((int)unadjustedPixelSize.Height, MinPixelHeight);
+            int resolution = WpfUnitsConversion.GetResolution(Content as Visual);
+
+            // Throttle rendering of plot while user is resizing the window.
+            // Plot rendering isn't fast enough to keep up with live resizing,
+            // which is what happens with undocked VS tool windows.
+            var model = Model;
+            if (model != null) {
+                _resizeAction.Invoke(() => model.ResizePlotAsync(pixelWidth, pixelHeight, resolution));
+            }
+        }
+
+        private void Image_MouseLeftButtonUp(object sender, MouseButtonEventArgs e) {
+            var image = (FrameworkElement)sender;
+            var pos = e.GetPosition(image);
+            var pixelSize = WpfUnitsConversion.ToPixels(image as Visual, pos);
+
+            Model?.ClickPlot((int)pixelSize.X, (int)pixelSize.Y);
+        }
+
+        private void UserControl_Drop(object sender, DragEventArgs e) {
+            var source = PlotClipboardData.Parse((string)e.Data.GetData(PlotClipboardData.Format));
+            if (source != null) {
+                bool isMove = (e.KeyStates & DragDropKeyStates.ShiftKey) != 0;
+                Model?.CopyPlotFromAsync(source.DeviceId, source.PlotId, isMove).DoNotWait();
+                e.Handled = true;
+            }
+        }
+
+        private void UserControl_DragEnter(object sender, DragEventArgs e) {
+            if (e.Data.GetDataPresent(PlotClipboardData.Format)) {
+                var source = PlotClipboardData.Parse((string)e.Data.GetData(PlotClipboardData.Format));
+                if (source != null) {
+                    var targetDeviceId = Model?.DeviceId;
+                    if (targetDeviceId != source.DeviceId) {
+                        if (source.ProcessId == Model?.SessionProcessId) {
+                            bool isMove = (e.KeyStates & DragDropKeyStates.ShiftKey) != 0;
+                            e.Effects = isMove ? DragDropEffects.Move : DragDropEffects.Copy;
+                            e.Handled = true;
+                            return;
+                        }
+                    }
+                }
+            }
+
+            e.Effects = DragDropEffects.None;
+            e.Handled = true;
+        }
+
+        private void UserControl_DragOver(object sender, DragEventArgs e) {
+            UserControl_DragEnter(sender, e);
+        }
+
+        private void Image_MouseMove(object sender, MouseEventArgs e) {
+            if (_dragSurface.IsMouseMoveStartingDrag(e)) {
+                var data = new DataObject(
+                    PlotClipboardData.Format,
+                    new PlotClipboardData(Model.DeviceId, Model.ActivePlotId, Model.SessionProcessId, false).ToString()
```

Comment:
No null check on `Model` here, like there is in other places where it's used. Is it certain to not be null at this point?